pkg/daemon/ceph/client/pool.go

```diff
@@ -447,6 +466,9 @@ func createReplicatedPoolForApp(context *clusterd.Context, clusterInfo *ClusterI
 		// Create the pool since it doesn't exist yet
 		// If there was some error other than ENOENT (not exists), go ahead and ensure the pool is created anyway
 		args := []string{"osd", "pool", "create", pool.Name, pgCount, "replicated", crushRuleName, "--size", strconv.FormatUint(uint64(pool.Replicated.Size), 10)}
+		if autoscaleMode != "" {
+			args = append(args, "--autoscale-mode", string(autoscaleMode))
+		}
```

Comment:
A particular issue I see with using the `--autoscale-mode` flag instead of the `pg_autoscale_mode` property is that properties are mutable and can be modified after object creation. The `--autoscale-mode` flag can only be set on CephBlockPool creation, which means it can only be selected when the pool is created. That actually creates a more challenging situation for users because they will need to migrate data to a new pool if they wish to update after initial creation. It would also mean that this PR is missing CRD validation logic that it is immutable and cannot be changed.

However, we agreed many weeks ago that the implementation should use a property unless a property doesn't exist. The `pg_autoscale_mode` property was undocumented by Ceph and was missed, but I pointed it out during my [last update](https://github.com/rook/rook/pull/11967#discussion_r1174439125). The property is also documented at the [link](https://docs.ceph.com/en/latest/rados/operations/placement-groups/#autoscaling-placement-groups) suggested by Parth [in this commit](https://github.com/rook/rook/pull/11967#issuecomment-1516539333). I continue to see evidence that this should be a documentation-only feature.

Reply:
Good catch, if we can use the `pg_autoscale_mode` property instead of `--autoscale-mode` flag, this will be preferred.

Reply:
I don't understand why "--autoscale-mode" that it can only be selected when pool is created?

Reply:
@zhucan The suggestion is to try this in the CephBlockPool parameters. Then we don't need to define the new setting in the CRD since the setting already exists as a pool parameter.
```yaml
  parameters:
    pg_autoscale_mode: off|on|warn
```

Reply:
@travisn  Thanks for youre explains.

Reply:
@travisn  done